tests/test_ballot_generator.py

```diff
@@ -0,0 +1,402 @@
+    }
+    bloc_voter_prop = {"W": 0.7, "C": 0.3}
+
+    # Find ballot probs
+    possible_rankings = it.permutations(candidates, ballot_length)
+    possible_ballots = [
+        Ballot(ranking=[{cand} for cand in ranking], weight=Fraction(1))
+        for ranking in possible_rankings
+    ]
+    ballot_prob_dict = {b: 0 for b in possible_ballots}
+
+    for b in possible_ballots:
+        ranking = b.ranking
+        for bloc in bloc_voter_prop.keys():
+            support_for_cands = pref_interval_by_bloc[bloc]
+            total_prob = 1
+            prob = bloc_voter_prop[bloc]
+            for cand in ranking:
+                prob *= support_for_cands[cand] / total_prob
+                total_prob -= support_for_cands[cand]
+            ballot_prob_dict[b] += prob
+
+    # Generate ballots
+    generated_profile = PlackettLuce(
+        number_of_ballots=number_of_ballots,
+        ballot_length=ballot_length,
+        candidates=candidates,
+        pref_interval_by_bloc=pref_interval_by_bloc,
+        bloc_voter_prop=bloc_voter_prop,
+    )
+
+    # Test
+    assert do_ballot_probs_match_ballot_dist(ballot_prob_dict, generated_profile)
+
+
+def test_BT_distribution():
+
+    # Set-up
+    number_of_ballots = 1000
+    ballot_length = 4
+    candidates = ["W1", "W2", "C1", "C2"]
+    ballot_length = None
+    pref_interval_by_bloc = {
+        "W": {"W1": 0.4, "W2": 0.3, "C1": 0.2, "C2": 0.1},
+        "C": {"W1": 0.2, "W2": 0.2, "C1": 0.3, "C2": 0.3},
+    }
+    bloc_voter_prop = {"W": 0.7, "C": 0.3}
+
+    # Find ballot probs
+    possible_rankings = it.permutations(candidates, ballot_length)
+    possible_ballots = [
+        Ballot(ranking=[{cand} for cand in ranking], weight=Fraction(1))
+        for ranking in possible_rankings
+    ]
+    ballot_prob_dict = {b: 0 for b in possible_ballots}
+
+    for b in possible_ballots:
+        ranking = b.ranking
+        for bloc in bloc_voter_prop.keys():
+            support_for_cands = pref_interval_by_bloc[bloc]
+            prob = bloc_voter_prop[bloc]
+            for i in range(len(ranking)):
+                greater_cand = support_for_cands[ranking[i]]
+                for j in range(i, len(ranking)):
+                    cand = support_for_cands[ranking[j]]
+                    prob *= greater_cand / (greater_cand + cand)
+            ballot_prob_dict[b] += prob
+
+        normalizer = 1 / sum(ballot_prob_dict.values())
+        ballot_prob_dict = {k: v * normalizer for k, v in ballot_prob_dict.items()}
+
+    # Generate ballots
+    generated_profile = BradleyTerry(
+        number_of_ballots=number_of_ballots,
+        ballot_length=ballot_length,
+        candidates=candidates,
+        pref_interval_by_bloc=pref_interval_by_bloc,
+        bloc_voter_prop=bloc_voter_prop,
+    )
+
+    # Test
+    assert do_ballot_probs_match_ballot_dist(ballot_prob_dict, generated_profile)
+
+
+def test_BT_probability_calculation():
+
+    # Set-up
```

Comment:
tip: this setup that gets copied and pasted could be refactored into a pytest.fixture: https://docs.pytest.org/en/7.1.x/how-to/fixtures.html (not necessary, but might be nice)